internal/auth/token.go

```diff
@@ -129,36 +140,50 @@ func (a Authorization) IssueToken(cfg keppel.Configuration) (*TokenResponse, err
 		Access:   a.ScopeSet.Flatten(),
 		Embedded: embeddedUserIdentity{UserIdentity: a.UserIdentity},
 	})
+	//we need to remember which key we used for this token, to choose the right
+	//key for validation during parseToken()
+	token.Header["jwk"] = serializePublicKey(issuerKey)
 
-	var (
-		jwkMessage json.RawMessage
-		err        error
-	)
-	jwkMessage, err = issuerKey.PublicKey().MarshalJSON()
-	if err != nil {
-		return nil, err
-	}
-	token.Header["jwk"] = &jwkMessage
-
-	signed, err := token.SignedString(issuerKey.CryptoPrivateKey())
+	tokenStr, err := token.SignedString(issuerKey)
 	return &TokenResponse{
-		Token:     signed,
+		Token:     tokenStr,
 		ExpiresIn: uint64(expiresAt.Sub(now).Seconds()),
 		IssuedAt:  now.Format(time.RFC3339),
 	}, err
 }
 
-func chooseSigningMethod(key libtrust.PrivateKey) jwt.SigningMethod {
-	issuerKey := key.CryptoPrivateKey()
-	switch issuerKey.(type) {
-	case *ed25519.PrivateKey:
+func chooseSigningMethod(key crypto.PrivateKey) jwt.SigningMethod {
+	switch key.(type) {
+	case ed25519.PrivateKey:
 		return jwt.SigningMethodEdDSA
-	case *ecdsa.PrivateKey:
-		return jwt.SigningMethodES256
 	case *rsa.PrivateKey:
 		return jwt.SigningMethodRS256
 	default:
-		panic(fmt.Sprintf("do not know which JWT method to use for issuerKey.type = %T", issuerKey))
+		panic(fmt.Sprintf("do not know which JWT method to use for issuerKey.type = %T", key))
+	}
+}
+
+func derivePublicKey(key crypto.PrivateKey) crypto.PublicKey {
+	switch key := key.(type) {
+	case ed25519.PrivateKey:
+		return key.Public()
+	case *rsa.PrivateKey:
+		return key.Public()
+	default:
+		panic(fmt.Sprintf("do not know which JWT method to use for issuerKey.type = %T", key))
+	}
+}
+
+func serializePublicKey(key crypto.PrivateKey) string {
+	switch key := key.(type) {
+	case ed25519.PrivateKey:
+		pubkey := key.Public().(ed25519.PublicKey)
+		return hex.EncodeToString([]byte(pubkey))
+	case *rsa.PrivateKey:
+		pubkey := key.Public().(*rsa.PublicKey)
+		return fmt.Sprintf("%x:%s", pubkey.E, pubkey.N.Text(16))
```

Comment:
```suggestion
		return fmt.Sprintf("%x:%x", pubkey.E, pubkey.N)
```
Why are we not doing this? Is there a catch here?

Reply:
This won't work because `pubkey.N` is a bigint. We need to use its custom methods for formatting.